icefoganalytics/elcc-data-management
# ELCC-4: Establish Appropriate Primary Keys In Database

Fixes ELCC-4

# Context

Add primary key to `user_roles` and `users` tables.
Add appropriate uniqueness constraint to `users.email`
Check for any other tables missing this kind of thing.

# Implemenation

Dropping constraints by config values doesn't seem to have very good support so I threw up a partial feature for MSSQL.

# Testing Instructions

1. Boot the app via `dev up`. This also check if the migrations will run, as they run on container boot.
2. Check that the app complies, and that you can log in at http://localhost:8080.
3. Check you can run the new migrations, and roll them back via
    ```bash
    dev migrate down -- --step 2
    dev migrate up
    ```
